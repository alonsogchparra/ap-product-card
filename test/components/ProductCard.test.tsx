import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ProductCard from '../../src/components';
import { product1 } from '../../src/data/products';

describe('ProductCard', () => {
  test('should render the component correctly', () => {
    const { asFragment } = render(
      <ProductCard product={product1}>
        {() => <h1>Product Card</h1>}
      </ProductCard>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should counter increment', () => {
    const { asFragment } = render(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>Increase</button>
          </>
        )}
      </ProductCard>
    );

    const countSpan = screen.getByText('0');
    const button = screen.getByRole('button', { name: 'Increase' });

    expect(countSpan.textContent).toBe('0');

    fireEvent.click(button);
    expect(countSpan.textContent).toBe('1');

    expect(asFragment()).toMatchSnapshot();
  });
});
