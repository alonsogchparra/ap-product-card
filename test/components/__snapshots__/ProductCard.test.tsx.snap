// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should counter increment 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
    <span>
      1
    </span>
    <button>
      Increase
    </button>
  </div>
</DocumentFragment>
`;

exports[`ProductCard should render the component correctly 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
  </div>
</DocumentFragment>
`;
